digits/model/tasks/caffe_train.py

```diff
@@ -860,7 +860,10 @@ def _pycaffe_args(self, gpu_id):
             solver_type = solver_type_mapping[self.solver_type]
         except KeyError:
             raise ValueError("Unknown solver type {}.".format(self.solver_type))
-        gpu_script = "caffe.set_device({id});".format(id=gpu_id) if gpu_id else ""
+        if gpu_id:
+            gpu_script = "caffe.set_device({id});caffe.set_mode_gpu();".format(id=gpu_id)
+        else:
+            gpu_script = "caffe.set_mode_gpu();"
```

Comment:
@lukeyeager , thanks for pointing that to me.  Setting the mode to CPU when gpu_id is None makes sense to me, too.